lib/cursor.js

```diff
@@ -0,0 +1,73 @@
+import {Meteor} from 'meteor/meteor';
+import {_} from 'meteor/underscore';
+
+import {BaseDocument} from './base';
+import {User} from './user';
+
+export class Cursor extends BaseDocument {
+  // createdAt: time of document creation
+  // author:
+  //   _id
+  //   username
+  //   avatar
+  // contentKey: ID used to identify editor's content
+  // clientId: ID of the client this cursor belongs to
+  // color: Generated hex color to show to use for this user
+  // ranges: Array of selection ranges made by the user
+  // head: Position of the document where the cursor is now placed
+
+  static REFERENCE_FIELDS() {
+    return {
+      _id: 1,
+    };
+  }
+
+  static PUBLISH_FIELDS() {
+    return _.extend(super.PUBLISH_FIELDS(), {
+      _id: 1,
+      createdAt: 1,
+      head: 1,
+      ranges: 1,
+      author: 1,
+      contentKey: 1,
+      color: 1,
+      clientId: 1,
+    });
+  }
+
+  getReference() {
+    return _.pick(this, Object.keys(this.constructor.REFERENCE_FIELDS()));
+  }
+}
+
+Cursor.Meta({
+  name: 'Cursor',
+  fields(fields) {
+    return _.extend(fields, {
+      author: Cursor.ReferenceField(User, User.REFERENCE_FIELDS()),
+    });
+  },
+});
+
+Cursor.update = function upsert(...args) {
+  args.unshift('Cursor.update');
+  return Meteor.call(...args);
+};
+
+Cursor.clear = function clear(...args) {
```

Comment:
Is this ever used by the client? It should probably never be, or at least the version used by the client should use "connection ID" based on the current connection calling the method and not be provided as an argument.

I suggest that the main code for this is server-only, and then if we need method we would call that server code.